cli/tests/pcluster/api/controllers/test_image_operations_controller.py

```diff
@@ -74,20 +75,142 @@ def test_describe_image(self, client):
         )
         assert_that(response.status_code).is_equal_to(200)
 
-    def test_describe_official_images(self, client):
-        """Test case for describe_official_images."""
+
+def _create_official_image_info(version, os, architecture):
+    return ImageInfo(
+        {
+            "Name": f"aws-parallelcluster-{version}-{OS_TO_IMAGE_NAME_PART_MAP[os]}-{architecture}-other",
+            "Architecture": "x86_64",
+            "ImageId": "ami-test",
+        }
+    )
+
+
+def _describe_official_images_expected_response(version, os, architecture):
+    return {
+        "amiId": "ami-test",
+        "os": os,
+        "name": f"aws-parallelcluster-{version}-{OS_TO_IMAGE_NAME_PART_MAP[os]}-{architecture}-other",
+        "architecture": architecture,
+    }
+
+
+class TestDescribeOfficialImages:
+    def _send_test_request(self, client, version=None, os=None, architecture=None, region="us-east-1"):
         query_string = [
-            ("version", "version_example"),
-            ("region", "eu-west-1"),
-            ("os", "os_example"),
-            ("architecture", "architecture_example"),
-            ("nextToken", "next_token_example"),
+            ("version", version),
+            ("region", region),
+            ("os", os),
+            ("architecture", architecture),
         ]
         headers = {
             "Accept": "application/json",
         }
-        response = client.open("/v3/images/official", method="GET", headers=headers, query_string=query_string)
-        assert_that(response.status_code).is_equal_to(200)
+        return client.open("/v3/images/official", method="GET", headers=headers, query_string=query_string)
+
+    @pytest.mark.parametrize(
+        "version, os, arch, mocked_response, expected_response",
+        [
+            pytest.param(
```

Comment:
nice I was not aware of pytest.param that lets you specify the id together with the params. much better!